core/whistle_services-1.0.0/src/wh_services.erl

```diff
@@ -794,56 +807,87 @@ calculate_services_charges(#wh_services{jobj=ServiceJObj}=Services, ServicePlans
+            wh_json:set_value(<<"activation_charges">>, Total+Amount, Acc);
+        MetaData ->
+            Category = wh_json:get_value(<<"category">>, MetaData),
+            Class = wh_json:get_value(<<"class">>, MetaData),
+            Amount = wht_util:units_to_dollars(wh_transaction:amount(Transaction)),
+            Total = wh_json:get_value(<<"activation_charges">>, Acc, 0),
+            wh_json:set_values([
+                {<<"activation_charges">>, Total+Amount}
+                ,{[Category, Class, <<"activation_charges">>], Amount}
+            ], Acc)
+    end.
 
 %%--------------------------------------------------------------------
-%% @public
+%% @private
 %% @doc
-%%
 %% @end
 %%--------------------------------------------------------------------
--spec dry_run(ne_binary(), [{ne_binary(), ne_binary(), integer()}]) -> wh_json:object().
--spec dry_run(ne_binary(), [{ne_binary(), ne_binary(), integer()}], any()) -> wh_json:object().
-dry_run(Account, Updates) ->
-    dry_run(Account, Updates, []).
-
-dry_run(Account, Updates, Transactions) ->
-    Services = fetch(Account),
-    UpdatedServices = dry_run_update(Updates, Services),
-    calculate_charges(UpdatedServices, Transactions).
-
--spec dry_run_update([{ne_binary(), ne_binary(), integer()}], services()) -> services().
-dry_run_update([], Services) -> Services;
-dry_run_update([{Category, Classification, Quantity}|Updates], Services) ->
-    NewServices = wh_services:update(Category, Classification, Quantity, Services),
-    dry_run_update(Updates, NewServices).
-
-%%%===================================================================
-%%% Internal functions
-%%%===================================================================
+-spec dry_run_activation_charges(services()) -> wh_transactions:transactions().
+-spec dry_run_activation_charges(ne_binary(), wh_json:object()
+                                 ,services(), wh_transactions:transactions()) -> wh_transactions:transactions().
+-spec dry_run_activation_charges(ne_binary(), ne_binary()
+                                 ,wh_json:object(), services()
+                                 ,wh_transactions:transactions()) -> wh_transactions:transactions().
+dry_run_activation_charges(#wh_services{updates=Updates}=Services) ->
+    wh_json:foldl(
+        fun(Category, CategoryJObj, Acc) ->
+            dry_run_activation_charges(Category, CategoryJObj, Services, Acc)
+        end
+        ,[]
+        ,Updates
+    ).
+
+dry_run_activation_charges(Category, CategoryJObj, Services, Transactions) ->
+    wh_json:foldl(
+        fun(Class, Quantity, Acc1) ->
+            dry_run_activation_charges(Category, Class, Quantity, Services, Acc1)
+        end
+        ,Transactions
+        ,CategoryJObj
+    ).
+
+dry_run_activation_charges(Category, Class, Quantity, #wh_services{jobj=JObj}=Services, Transactions) ->
+    OldQ = wh_json:get_value([?QUANTITIES, Category, Class], JObj, 0),
+    case OldQ =:= Quantity of
```

Comment:
Since you don't use OldQ anywhere else, I would just do:

```erlang
case wh_json:get_value([...], JObj, 0) of
    Quantity -> Transactions;
    _OldQuantities -> ...
end.
```